File: Electronics_Shop_17.Model/Requests/AddProduct.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Numerics;
using System.Text;
using System.Threading.Tasks;
using Electronics_Shop_17.Model.Helpers;

namespace Electronics_Shop_17.Model.Requests
{
    public class AddProduct
    {
        public string Brand { get; set; }
        public string Model { get; set; }
        public string Description { get; set; }
        public decimal Price { get; set; }
        public int AllColorsStock { get; set; }

        public int ProductCategoryId { get; set; }
        public virtual List<AddProductProductTag>? ProductProductTags { get; set; }
        public virtual AddWarranty? Warranty { get; set; }


        public virtual AddCamera? Camera { get; set; }
        public virtual AddDesktopPC? DesktopPC { get; set; }
        public virtual AddGamingConsole? GamingConsole { get; set; }
        public virtual AddLaptop? Laptop { get; set; }
        public virtual AddPhone? Phone { get; set; }
        public virtual AddTablet? Tablet { get; set; }
        public virtual AddTelevision? Television { get; set; }

        public virtual AddAccessory? Accessory { get; set; }
        public bool isDeleted { get; set; } = false;

        public string StateMachine { get; set; } = "Initial";
    }
}

File: angular17_web/src/app/models/product/add.product.ts
import {AddProductColor} from '../product-color/add.product.color';
import {AddWarranty} from '../warranty/add.warranty';
import {AddDesktopPC} from '../desktop-pc/add.desktopPC';
import {AddCamera} from '../camera/add.camera';
import {AddGamingConsole} from '../gaming-console/add.gaming.console';
import {AddLaptop} from '../laptop/add.laptop';
import {AddPhone} from '../phone/add.phone';
import {AddTablet} from '../tablet/add.tablet';
import {AddTelevision} from '../television/add.television';
import {AddAccessory} from '../accessory/add.accessory';
import {AddProductProductTag} from '../product-product-tag/add.product.product.tag';

export class AddProduct {
  brand!: string;
  model!: string;
  description!: string;
  price!: number;

  productCategoryId!: number;
  productProductTags?: AddProductProductTag[];
  productColors!: AddProductColor[];
  warranty?: AddWarranty;

  camera?: AddCamera;
  desktopPC?: AddDesktopPC;
  gamingConsole?: AddGamingConsole;
  laptop?: AddLaptop;
  phone?: AddPhone;
  tablet?: AddTablet;
  television?: AddTelevision;

  accessory?: AddAccessory;
  isDeleted?: boolean;
  stateMachine?: string;

  constructor(data?: any) {
    if (!data) return;

    this.brand = data.brand;
    this.model = data.model;
    this.description = data.description;
    this.price = data.price;

    this.productCategoryId = data.productCategoryId;
    this.productProductTags = Array.isArray(data.productProductTags)
      ? data.productProductTags.map((x: any) => new AddProductProductTag(x))
      : undefined;
    this.productColors = Array.isArray(data.productColors)
      ? data.productColors.map((x: any) => new AddProductColor(x))
      : [];
    this.warranty = data.warranty ? new AddWarranty(data.warranty) : undefined;

    this.camera = data.camera ? new AddCamera(data.camera) : undefined;
    this.desktopPC = data.desktopPC ? new AddDesktopPC(data.desktopPC) : undefined;
    this.gamingConsole = data.gamingConsole ? new AddGamingConsole(data.gamingConsole) : undefined;
    this.laptop = data.laptop ? new AddLaptop(data.laptop) : undefined;
    this.phone = data.phone ? new AddPhone(data.phone) : undefined;
    this.tablet = data.tablet ? new AddTablet(data.tablet) : undefined;
    this.television = data.television ? new AddTelevision(data.television) : undefined;

    this.accessory = data.accessory ? new AddAccessory(data.accessory) : undefined;
    this.isDeleted = data.isDeleted;
    this.stateMachine = data.stateMachine;
  }
}
